File: server_integration_tests/clients/dart_client/test/fixtures.dart
import 'dart:io';
import 'package:graphlink_server_integration_tests_dart_client/generated/graphlink.dart';

int get _port => int.tryParse(Platform.environment['SERVER_PORT'] ?? '') ?? 9997;

String get httpUrl => 'http://localhost:$_port/graphql';
String get wsUrl => 'ws://localhost:$_port/graphql';

DefaultGraphLinkWebSocketAdapter newWsAdapter() =>
    DefaultGraphLinkWebSocketAdapter(url: wsUrl, reconnect: true, protocols: ['graphql-transport-ws']);

GraphLinkClient newClient() => GraphLinkClient(
      adapter: GraphLinkDioAdapter(url: httpUrl).call,
      wsAdapter: newWsAdapter(),
    );

File: server_integration_tests/clients/swift_client/Tests/GraphLinkGeneratedTests/Fixtures.swift
import Foundation
@testable import GraphLinkGenerated

enum Fixtures {
    static var port: Int {
        ProcessInfo.processInfo.environment["SERVER_PORT"].flatMap(Int.init) ?? 9997
    }

    static var SERVER_URL: URL { URL(string: "http://localhost:\(port)/graphql")! }
    static var WS_URL: URL { URL(string: "ws://localhost:\(port)/graphql")! }

    static func newClient() -> GraphLinkClient {
        GraphLinkClient.create(url: SERVER_URL, wsUrl: WS_URL)
    }
}
